sigstore-java/src/main/java/dev/sigstore/KeylessSigner.java

```diff
@@ -199,12 +285,49 @@ public List<KeylessSigningResult> sign(List<byte[]> artifactDigests)
     return result.build();
   }
 
+  private void renewSigningCertificate()
+      throws InterruptedException, CertificateException, IOException, UnsupportedAlgorithmException,
+          NoSuchAlgorithmException, InvalidKeyException, SignatureException,
+          FulcioVerificationException, OidcException {
+    lock.writeLock().lock();
+    try {
+      if (signingCert != null) {
+        @SuppressWarnings("JavaUtilDate")
+        long lifetimeLeft =
+            signingCert.getLeafCertificate().getNotAfter().getTime() - System.currentTimeMillis();
+        if (lifetimeLeft > minSigningCertificateLifetime.toMillis()) {
+          // The current certificate is fine, reuse it
+          return;
```

Comment:
I'm being pretty nitpicky here, but I believe this can be in a read lock? and then everything after can be in a write lock?

Reply:
I have the same thoughts.